python/custom_test_files.py

```diff
@@ -0,0 +1,189 @@
+            headers = json.load(d)
+            headers = pd.DataFrame(headers)
+        #Reads in the data from lar_file. 
+        df = pd.read_csv(lar_file, delimiter="|", header=None, dtype='object',
+                         keep_default_na=False) 
+        
+        df.columns = headers['field'] #Takes the field names from the LAR schema as column names.
+        
+        current_row = len(df.index) #Number of rows currently in the dataframe. 
+
+        #Calculates a multiplier taking the ceiling function of desired row count over current row count. 
+        multiplier = math.ceil(row_count/current_row)
+        
+        #Concatenates data to produce the number of rows by the multiplier in a new LAR dataframe. 
+        df = pd.concat([df]*int(multiplier))
+        
+        #Drops the number of rows to the row count specified. 
+        
+        if (row_count%current_row) == 0:
+            return df
+        else:
+            drop_rows = current_row - (row_count%current_row)
+            df = df[:-(drop_rows)]
+            return df
+         
+    def assign_uli(self, df=None, ts_file=None):
+        """
+        Assigns new ULI's to a dataframe of lar rows, based on the bank number in the TS file. 
+        
+        """
+        #Reads in ts_file.
+        ts_df = pd.read_csv(ts_file, delimiter="|", header=None, dtype='object',
+                         keep_default_na=False)
+        
+        #Stores the second column of the TS file as 'bank'. 
+        bank = ts_df.iloc[0][1]
+        
+        #Assigns the number of rows in the dataframe to row_count. 
+        df.reset_index()
+        row_count = len(df.index)
+        #Creates a new list of LEI's for the ULI column by bank.
+        if bank == 'Bank0': 
+            unique_lei_for_uli = {'uli':list(['B90YWS6AFX2LGWOXJ1LD']*int((row_count)))}
+        
+        if bank == 'Bank1':
+            unique_lei_for_uli = {'uli':list(['BANK1LEIFORTEST12345']*int((row_count)))}
+        
+        uli_col = pd.DataFrame(unique_lei_for_uli) #Creates a dataframe of new ULI's.
+        #Amends the ULI's by adding up to 23 additional characters. 
+        uli_col['uli'] = uli_col.apply(lambda x: x.uli + lg.char_string_gen(length = 23), axis=1)
+        #Adds a checkdigit to the end of the ULI. 
+        uli_col['uli'] = uli_col.apply(lambda x: x.uli + lg.check_digit_gen(ULI=x.uli), axis=1)
+        #Checks whether ULI's are unique using the uli_check function. 
+        
+        if self.uli_check(col = uli_col) is True:
+            df['uli'] = uli_col['uli'] #If unique, ULI's in the dataframe are replaced. 
+            print("Unique ULIs Assigned for " + str(bank))
+            return df
+        else: 
+            #If not unique, the assign_uli function is re-run.
+            print("Re-Running")
+            assign_uli(self, ts_file=ts_file, df=df)  
+    
+    def uli_check(self, col=None):
```

Comment:
I think this function may be unnecessary. 

There's a couple ways you could handle this logic inside the assign_uli function. 

1) Check if the Set() of ULIs is equal to the len() of the ULIs.
2) Use the unique function from dataframe/dataseries to see if the length of unique values is the same as the length of total values.